crates/uv/src/commands/project/run.rs

```diff
@@ -363,35 +400,28 @@ pub(crate) async fn run(
             false,
         )?;
 
-        if requirements.is_empty() {
-            Some(venv)
-        } else {
-            debug!("Syncing ephemeral requirements");
-
-            let client_builder = BaseClientBuilder::new()
-                .connectivity(connectivity)
-                .native_tls(native_tls);
-
-            let spec =
-                RequirementsSpecification::from_simple_sources(&requirements, &client_builder)
-                    .await?;
-
-            // Install the ephemeral requirements.
-            Some(
-                project::update_environment(
-                    venv,
-                    spec,
-                    &settings,
-                    &state,
-                    preview,
-                    connectivity,
-                    concurrency,
-                    native_tls,
-                    cache,
-                    printer,
+        match spec {
+            None => Some(venv),
+            Some(spec) if spec.is_empty() => Some(venv),
```

Comment:
You might be able to simplify this with a `spec.filter(|spec| !spec.is_empty())` somewhere.